extensions/replication/utils/utils.js

```diff
@@ -0,0 +1,17 @@
+const { encode } = require('arsenal').versioning.VersionID;
+const ObjectMD = require('arsenal').models.ObjectMD;
+
+function getVersionIdForReplication(sourceEntry, encoding = false) {
+    const nullCheckFn = sourceEntry instanceof ObjectMD ? entry => entry.getIsNull() : entry => entry.isNull;
+    const getVIDFn = sourceEntry instanceof ObjectMD ? entry => entry.getVersionId() : entry => entry.versionId;
```

Comment:
what other type can it be?
it would be better to just add the `getIsNull()` and `getVersionId()` methods in that type...

Reply:
Also, this replaces calls to `getEncodedVersionId()`, which is implemented in some (or all?) the types....

*To be confirmed*, but it looks to me like this logic should actually be in the getEncodedVersionId() function [from arsenal ObjectMD)?

Reply:
Indeed the changes can be simplified a lot just by updating `getEncodedVersionId()` to handle the versioning suspended case.

This is done in the following PR: https://github.com/scality/Arsenal/pull/2033